tarpc/src/transport/tcp.rs

```diff
@@ -0,0 +1,62 @@
+use std::io;
+use std::net::{SocketAddr, TcpListener, TcpStream, ToSocketAddrs};
+use std::time::Duration;
+
+/// A transport for TCP.
+pub struct TcpTransport<A: ToSocketAddrs>(pub A);
+impl<A: ToSocketAddrs> super::Transport for TcpTransport<A> {
+    type Listener = TcpListener;
+    fn bind(&self) -> io::Result<TcpListener> {
+        TcpListener::bind(&self.0)
+    }
+}
+
+impl<A: ToSocketAddrs> super::Transport for A {
+    type Listener = TcpListener;
+    fn bind(&self) -> io::Result<TcpListener> {
+        TcpListener::bind(self)
+    }
+}
+
+impl super::Listener for TcpListener {
+    type Dialer = TcpDialer<SocketAddr>;
+    type Stream = TcpStream;
+    fn accept(&self) -> io::Result<TcpStream> {
+        self.accept().map(|(stream, _)| stream)
```

Comment:
The dialer is available on the listener though.
On Mar 2, 2016 9:31 PM, "shaladdle" <notifications@github.com> wrote:

> In tarpc/src/transport/tcp.rs
> <https://github.com/google/tarpc/pull/27#discussion_r54837715>:
>
> > +        TcpListener::bind(&self.0)
> > +    }
> > +}
> > +
> > +impl<A: ToSocketAddrs> super::Transport for A {
> > +    type Listener = TcpListener;
> > +    fn bind(&self) -> io::Result<TcpListener> {
> > +        TcpListener::bind(self)
> > +    }
> > +}
> > +
> > +impl super::Listener for TcpListener {
> > +    type Dialer = TcpDialer<SocketAddr>;
> > +    type Stream = TcpStream;
> > +    fn accept(&self) -> io::Result<TcpStream> {
> > +        self.accept().map(|(stream, _)| stream)
>
> Did you think about converting the second element of the tuple to a Dialer
> or something? It seems conceivable that we might want this functionality
> later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/tarpc/pull/27/files#r54837715>.
>